app/src/main/java/com/example/xpvehicles/adapters/ExploreAdapter.java

```diff
@@ -65,14 +60,16 @@ public int getItemCount() {
         return vehicles.size();
     }
 
-    public void addAll(List<Vehicle> allVehicles) {
-        vehicles.addAll(allVehicles);
-        notifyDataSetChanged();
-    }
-
-    public void clear() {
+    public void setVehicles(List<Vehicle> allVehicles, TextView textViewNoVehicles) {
         vehicles.clear();
         notifyDataSetChanged();
+        if (allVehicles.size() > 0) {
+            textViewNoVehicles.setVisibility(View.GONE);
+            vehicles.addAll(allVehicles);
+            notifyDataSetChanged();
```

Comment:
I think we should only clear  if it is larger than 0?

Reply:
If I only clear if larger than 0, it causes a bug where in explore fragment it displays "no vehicles for rent" without removing any of the previous vehicles